### PR TITLE
More performance improvement, mostly on the action=count view:

### PR DESCRIPTION
**Pull request by pascalchevrel**
- DotLangParser::extract_metadata static boolean property, allows a view to not extract any metadata when parsing lang files
- LangManager::error_checking static boolean property, allows a view to not look for errors in strings when parsing lang files
- Update main loop in DotLangParser::ParseFile() to test most common case first (translation exists)
- many small nitpicks and cleanups

Performance gains are mostly on the 'count' view (40% for me) but other views also benefit from a small speedup because of the code reorganization (~10%).

### CODE REVIEW

**Comment by pascalchevrel** [comment]
Failing test, didn't notice, will update the PR soon

File: classes/Langchecker/DotLangParser.php [hunk 1]
```diff
@@ -29,10 +43,10 @@ class DotLangParser
      * @return mixed Cleaned up array of lines (array),
```
Comment (flodolo) [inline classes/Langchecker/DotLangParser.php]: Remove show_errors from method doc.

File: classes/Langchecker/DotLangParser.php [hunk 2]
```diff
@@ -127,24 +145,31 @@ public static function parseFile($path, $reference_locale = false)
...
                     }
 
-                    // Store the translation
-                    $dotlang_data['strings'][$reference] = $translation;
-
-                    $i++;
+                    /* Empty translation: what I'm reading as translation is either the next reference string
+                     * or the next meta tag (comment, tag binding). I consider this string untranslated.
+                     */
+                    $dotlang_data['strings'][$reference] = $reference;
```
Comment (flodolo) [inline classes/Langchecker/DotLangParser.php]: Completely missed this in the previous update :-\

File: classes/Langchecker/DotLangParser.php [hunk 3]
```diff
@@ -127,24 +145,31 @@ public static function parseFile($path, $reference_locale = false)
...
-                        $dotlang_data['strings'][$reference] = $reference;
+                    // If we don't extract metadata, we shouldn't test for #
+                    $markers = self::$extract_metadata
+                        ? [';', '#']
+                        : [';'];
+
+                    // We test the most common scenario first: reference string followed by a translation
+                    if (! Utils::startsWith($translation, $markers)) {
+                        // Store the translation
+                        $dotlang_data['strings'][$reference] = $translation;
```
Comment (flodolo) [inline classes/Langchecker/DotLangParser.php]: Should we set `$translation` in this conditional, considering that it's now used only here?

**Comment by flodolo** [comment]
Besides the comment, looks good to me.

File: classes/Langchecker/DotLangParser.php (hunk 3 above)
Comment (pascalchevrel) [inline classes/Langchecker/DotLangParser.php]: it needs to be set before the conditionnal because the if() close makes a test on the $translation variable

Comment (flodolo) [inline classes/Langchecker/DotLangParser.php]: Right, missed the previous line.

**Pull request merged**